### ISSUE TITLE
Enable python and django monitoring in kafka consumers and celery workers

### ISSUE THREAD

**Issue by robrap** [edx/edx-arch-experiments#785]
This probably should be two separate tasks (kafka consumers and celery workers), but adding one to start to capture the ideas.

Note: in Datadog, I'm not sure if the Python version can be seen manually. This might need to be asked in a DD support ticket.

Implementation thoughts:
- Take the part of DeploymentMonitoringMiddleware that actually sets the custom attributes, and make it into a utility method that could be used from outside the middleware as well
- In event-bus-kafka, we could [add a call to this new method here](https://github.com/openedx/event-bus-kafka/blob/d92281dc9e42d79e9303c246d0771ab00793aac2/edx_event_bus_kafka/internal/consumer.py#L559). Reminder: add an edx-django-utils minimum constraint in base.in to ensure it gets published as part of the package.
- The Celery change is a little trickier, only because we don't have a mechanism set up to plug in to all services yet (although this is very close). 
  - It would probably need to be implemented in a plugin. It would either use a celery signal or DD hook as mentioned in https://github.com/edx/edx-arch-experiments/issues/729.
  - If the plugin were defined in edx-django-utils, it would get added to all services (once the new plugin endpoint were defined and added to each service), because this library is already installed in each service.
    - The trouble with doing this in edx-django-utils is that it probably won't work for New Relic. That said, the feature benefits the entire community for upgrades, so hopefully this would be acceptable.
    - The alternative would be to create a 2u-django-utils repo that privately gets deployed to all our services. This is a nice-to-have capability in the long run, but would take additional work, so we'd need to consider if it is worth it.

**Comment by robrap** [comment]
@robrap: Note to self: Determine if any of this is still valid, or if it is all complete, and then clean-up this ticket.

**Comment by robrap** [comment]
@UsamaSadiq: You are welcome to take this. I will only note that this ticket is from early in the DD migration, so some of it is no longer relevant. As a first step, please help determine what is actually still needed, if anything. Thank you!

**Comment by UsamaSadiq** [comment]
`DatadogBackend` was verified to be present in all services and the Datadog migration tracking [sheet](https://docs.google.com/spreadsheets/d/1sh9eE-vT5JysI4BAx_qlkhzZ0hEujM_B18qL4XQoq1k/edit?gid=56476385#gid=56476385) was updated in the [issue](https://github.com/edx/edx-arch-experiments/issues/815).

**Comment by UsamaSadiq** [comment]
> Take the part of DeploymentMonitoringMiddleware that actually sets the custom attributes, and make it into a utility method that could be used from outside the middleware as well

The function [_set_custom_attribute()](https://github.com/openedx/edx-django-utils/blob/3f295d4f8e40a22f2b40db0bbabb0f6b2ba72406/edx_django_utils/monitoring/internal/middleware.py#L192) which is used within the middleware to record data is already defined outside the middleware but can't be defined as a public method due to circular dependencies issue.

**Comment by UsamaSadiq** [comment]
> In event-bus-kafka, we could [add a call to this new method here](https://github.com/openedx/event-bus-kafka/blob/d92281dc9e42d79e9303c246d0771ab00793aac2/edx_event_bus_kafka/internal/consumer.py#L559). Reminder: add an edx-django-utils minimum constraint in base.in to ensure it gets published as part of the package.


The public method [set-custom-attribute](https://github.com/openedx/edx-django-utils/blob/3f295d4f8e40a22f2b40db0bbabb0f6b2ba72406/edx_django_utils/monitoring/utils.py#L36) is already being [used](https://github.com/openedx/event-bus-kafka/blob/d92281dc9e42d79e9303c246d0771ab00793aac2/edx_event_bus_kafka/internal/consumer.py#L14C75-L14C95) inside event-bus-kafka. 
Creating PR to add the function calls to record python and django version from the celery workers.

**Comment by robrap** [comment]
@UsamaSadiq: What would it take to close this ticket? Thanks.

**Comment by robrap** [comment]
@UsamaSadiq: Can you summarize next steps on this ticket. Thank you.

**Comment by UsamaSadiq** [comment]
This ticket had two distinct parts in it. 
First one was to add support of Python/Django version monitoring in the kafka consumers and second part was to find a way to implement this for celery workers too. 
The first phase was completed with the merging of my PR and kafka consumers now have the support of logging Python/Django versions. 
The second part of adding this support in celery workers need a discussion on which approach to be used for it. I'd suggest creating a follow up issue separate to this ticket to track the second phase of the ticket since it'll require discussions and discovery for celery workers.
